### PR TITLE
Shivam/make job trigger configurable

### PR DESCRIPTION
**Pull request by shivamcausecode**

### CODE REVIEW

**Comment by shivamcausecode** [comment]
[(8) SPYD - Handle update of UFile and reflect the same in Elasticsearch index #200](https://trello.com/c/M1a2CtYR/348-8-spyd-handle-update-of-ufile-and-reflect-the-same-in-elasticsearch-index-200)

File: changelog.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # ChangeLog
 
+## Version 3.1.2 [03-April-2018]
```
Comment (ankit-agrawal11) [inline changelog.md]: https://github.com/causecode/grails-file-uploader/pull/30/files#diff-42ba1d994f4fa8c6ad17a7efae7936ccR1

Add changelog for 3.1.0

File: changelog.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+
+### Added
+- Made trigger of auto renew URL job configurable. Developer can disable the execution of the job from the installing app.
```
Comment (ankit-agrawal11) [inline changelog.md]: Functionality to disable URL renew job for installing application.

File: grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy [hunk 3]
```diff
@@ -20,17 +22,23 @@ class DailyJob {
     }
 
     def execute() {
-        log.info 'Started executing DailyJob..'
-        fileUploaderService.renewTemporaryURL()
-        fileUploaderService.moveFailedFilesToCDN()
+
+        boolean renewJobDisabled = Holders.config.jobs.fileUploader.renewURLs.disable ?: false
```
Comment (ankit-agrawal11) [inline grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy]: disable -> disabled
renewJobDisabled -> isDisabled

File: grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy [hunk 4]
```diff
@@ -20,17 +22,23 @@ class DailyJob {
...
+
+        if (!renewJobDisabled) {
+            log.info 'Started executing DailyJob..'
+
+            fileUploaderService.renewTemporaryURL()
+            fileUploaderService.moveFailedFilesToCDN()
+
+            log.info 'Finished executing DailyJob.'
+        }
```
Comment (ankit-agrawal11) [inline grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy]: Add else statement and log that the job has been disabled in installing app.

File: README.md [hunk 5]
```diff
@@ -118,4 +119,13 @@ Second flag will tell plugin which algorithm to use to calculate the checksum. C
...
 Algorithm.SHA1
 ```
-Second flag will be ignored if first flag is set to false. By default checksum calculation features is disabled. To enable provide valid configurations. 
\ No newline at end of file
+Second flag will be ignored if first flag is set to false. By default checksum calculation features is disabled. To enable provide valid configurations. 
+
+5. To disable the auto renew URL job from the installing app, add the following config in the application.groovy file.
+```
+jobs {
+    fileUploader.renewURLs.disable = true // By default it is set to false
```
[the remark this answers is not in the document]
Comment (shivamcausecode) [reply README.md]: No changes required. It is better than disabled as we are assigning.

File: README.md [hunk 6]
```diff
@@ -118,4 +119,13 @@ Second flag will tell plugin which algorithm to use to calculate the checksum. C
...
+}
+```
+
+6. To disable the expiration of the URL, Add `makePublic: true` to group config in application.groovy.
```
[the remark this answers is not in the document]
Comment (shivamcausecode) [reply README.md]: Changed

File: changelog.md (hunk 1 above)
Comment (shivamcausecode) [reply changelog.md]: Ok

File: changelog.md (hunk 2 above)
Comment (shivamcausecode) [reply changelog.md]: Changed

File: grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy (hunk 3 above)
Comment (shivamcausecode) [reply grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy]: isRenewUrlJobDisabled was better if we using a method here.

File: grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy (hunk 4 above)
Comment (shivamcausecode) [reply grails-app/jobs/com/causecode/fileuploader/DailyJob.groovy]: Ok

File: changelog.md [hunk 7]
```diff
@@ -1,5 +1,23 @@
 # ChangeLog
 
+## Version 3.1.2 [Unreleased]
+
+### Added
+- Functionality to disable URL renew job for installing application.
+
+#Version 3.1.1 [30-March-2018]
```
[the remark this answers is not in the document]
Comment (shivamcausecode) [reply changelog.md]: Removed

**Pull request merged**